pingcap/tidb
# set global variable tidb_txn_mode to pessimistic during bootstrap.

### What problem does this PR solve? 
Make the default `tidb_txn_mode` to `pessimistic` but do not affects the existing clusters.

### What is changed and how it works?
Set the global variable `tidb_txn_mode` to `pessimistic` during bootstrap.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
1. start an old version cluster, upgrade the cluster to this version, verify that `@@global.tidb_txn_mode` is still empty string.
2. clear data and start a new cluster of this version, verify that `@@global.tidb_txn_mode` is `pessimistic`.

Related changes

 - Need to cherry-pick to the release branch

Release note

 - set the initial global variable value of `tidb_txn_mode` to `pessimistic` during bootstrap.
